etcdmgmt/etcd-mgmt.go

```diff
@@ -160,3 +151,114 @@ func isETCDStartNeeded() (bool, int) {
 	}
 	return start, pid
 }
+
+// initETCDArgVar() will initialize etcd argument which will be used at various places
+func initETCDArgVar() {
+	HostIP, err := utils.GetLocalIP()
+	if err != nil {
+		log.Fatal("Could not able to get IP address")
+	}
+
+	context.HostIP = HostIP
+
+	listenClientUrls = "http://" + context.HostIP + ":2379"
+	advClientUrls = "http://" + context.HostIP + ":2379"
+	listenPeerUrls = "http://" + context.HostIP + ":2380"
+	initialAdvPeerUrls = "http://" + context.HostIP + ":2380"
+}
+
+// ETCDStartInit() Check whether etcd environment variable present or not
+// If it present then start etcd without --initial-cluster flag
+// other wise start etcd normally.
+func ETCDStartInit() (*os.Process, error) {
+	initETCDArgVar()
+
+	file, err := os.Open(etcdConfFile)
+	if err != nil {
+		switch {
+		case os.IsNotExist(err):
+			log.Info("Starting/Restarting etcd for a initial node")
+			return StartStandAloneETCD()
+		default:
+			log.WithFields(log.Fields{
+				"error": err,
+				"path":  etcdPidFile,
+			}).Fatal("Failed to read from file")
+		}
+	} else {
+		defer file.Close()
+
+		// Restoring etcd environment variable and starting etcd daemon
+		scanner := bufio.NewScanner(file)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			linestr := scanner.Text()
+			etcdenv := strings.Split(linestr, "=")
+
+			etcdEnvKey := etcdenv[0]
+			etcdEnvData := etcdenv[1]
+			envlen := len(etcdenv)
+			for i := 2; i < envlen; i++ {
+				etcdEnvData = etcdEnvData + "=" + etcdenv[i]
+			}
+
+			// setting etcd environment variable
+			os.Setenv(etcdEnvKey, etcdEnvData)
+		}
+
+		args := []string{"-listen-client-urls", listenClientUrls,
+			"-advertise-client-urls", advClientUrls,
+			"-listen-peer-urls", listenPeerUrls,
+			"-initial-advertise-peer-urls", initialAdvPeerUrls}
+
+		log.Info("Sstarting etcd daemon")
+		return StartETCD(args)
+	}
+	return nil, err
+}
+
+//StartStandAloneETCD() will Start default etcd by considering single server node
+func StartStandAloneETCD() (*os.Process, error) {
+	args := []string{"-listen-client-urls", listenClientUrls,
```

Comment:
its not good to call every time getLocalIP(). we just need to call initETCDArgVar at the time of starting glusterd. it will save call request.